lib/min_heap.rb

```diff
@@ -14,18 +14,29 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: ? O(logn)
+  # Space Complexity: ? O(1)
   def add(key, value = key)
```

Comment:
👍  However since you are doing recursion the space complexity is O(log n) due to the call stack.